scripts/run-android.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -m # needed to access jobs
+
+GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
+
+# generates android debug build when we pass the debug flag
+"${GIT_ROOT}"/scripts/build-android.sh
+
+# installs the resulting apk on running emulator or android device
+adb install ./result/app-debug.apk
+
+# start metro and go to background
+nohup "${GIT_ROOT}"/scripts/start-react-native.sh 2>&1 | tee metro-server-logs.log &
+
+echo "Waiting for Metro server to be ready..."
+while ! grep -q "Welcome to Metro" metro-server-logs.log; do
+  echo -n "."
+  sleep 1
+done
+
+# start the installed app
+adb shell monkey -p im.status.ethereum.debug 1
```

Comment:
Shouldn't we have a `trap` for when any of the `adb` commands fail to stop the `scripts/start-react-native.sh` script running in the background?

Reply:
the `adb` commands shouldn't stop the script running in background.
the background script is the metro server which I bring to foreground later in this file.

This `adb monkey` command is strategically placed here so that I start the installed app after I read from metro logs that the metro server is ready.

Reply:
I think Jakub means, that if smth goes wrong with any further command - we need to catch it by `trap` and stop the metro server, otherwise it might be left running.

Reply:
Oh thanks for explaining, I'll look into it